File: internal/domain/listing.go
package domain

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type Listing struct {
	ID         uuid.UUID `json:"id" db:"id"`
	SourceID   uuid.UUID `json:"source_id" db:"source_id"`
	ExternalID string    `json:"external_id" db:"external_id"`
	URL        string    `json:"url" db:"url"`

	// Core fields
	Title       string `json:"title" db:"title"`
	Description string `json:"description,omitempty" db:"description"`
	AskingPrice *int64 `json:"asking_price,omitempty" db:"asking_price"` // cents
	Revenue     *int64 `json:"revenue,omitempty" db:"revenue"`           // cents, annual
	CashFlow    *int64 `json:"cash_flow,omitempty" db:"cash_flow"`       // cents, annual (SDE/EBITDA)
	EBITDA      *int64 `json:"ebitda,omitempty" db:"ebitda"`             // cents
	Inventory   *int64 `json:"inventory_value,omitempty" db:"inventory_value"`

	// Real estate
	RealEstateIncluded bool   `json:"real_estate_included" db:"real_estate_included"`
	RealEstateValue    *int64 `json:"real_estate_value,omitempty" db:"real_estate_value"`

	// Location
	City     string   `json:"city,omitempty" db:"city"`
	State    string   `json:"state,omitempty" db:"state"`
	ZipCode  string   `json:"zip_code,omitempty" db:"zip_code"`
	Country  string   `json:"country" db:"country"`
	Lat      *float64 `json:"lat,omitempty" db:"lat"`
	Lng      *float64 `json:"lng,omitempty" db:"lng"`

	// Business details
	Industry         string `json:"industry,omitempty" db:"industry"`
	IndustryCategory string `json:"industry_category,omitempty" db:"industry_category"`
	BusinessType     string `json:"business_type,omitempty" db:"business_type"`
	YearEstablished  *int   `json:"year_established,omitempty" db:"year_established"`
	Employees        *int   `json:"employees,omitempty" db:"employees"`
	ReasonForSale    string `json:"reason_for_sale,omitempty" db:"reason_for_sale"`

	// Lease
	LeaseExpiration *time.Time `json:"lease_expiration,omitempty" db:"lease_expiration"`
	MonthlyRent     *int64     `json:"monthly_rent,omitempty" db:"monthly_rent"`

	// Franchise
	IsFranchise   bool   `json:"is_franchise" db:"is_franchise"`
	FranchiseName string `json:"franchise_name,omitempty" db:"franchise_name"`

	// Raw data
	RawData json.RawMessage `json:"raw_data,omitempty" db:"raw_data"`

	// Metadata
	FirstSeenAt time.Time `json:"first_seen_at" db:"first_seen_at"`
	LastSeenAt  time.Time `json:"last_seen_at" db:"last_seen_at"`
	IsActive    bool      `json:"is_active" db:"is_active"`
}

type ListingSearchParams struct {
	Query       string   `json:"q"`
	PriceMin    *int64   `json:"price_min"`
	PriceMax    *int64   `json:"price_max"`
	RevenueMin  *int64   `json:"revenue_min"`
	CashFlowMin *int64   `json:"cash_flow_min"`
	States      []string `json:"states"`
	Industries  []string `json:"industries"`
	Franchise   *bool    `json:"franchise"`
	RealEstate  *bool    `json:"real_estate"`
	Bounds      *GeoBounds `json:"bounds"`
	Sort        string   `json:"sort"`
	Page        int      `json:"page"`
	PerPage     int      `json:"per_page"`
}

type GeoBounds struct {
	SouthLat float64 `json:"south_lat"`
	WestLng  float64 `json:"west_lng"`
	NorthLat float64 `json:"north_lat"`
	EastLng  float64 `json:"east_lng"`
}

type ListingSearchResult struct {
	Listings   []Listing `json:"listings"`
	Total      int       `json:"total"`
	Page       int       `json:"page"`
	PerPage    int       `json:"per_page"`
	TotalPages int       `json:"total_pages"`
}

type FilterOptions struct {
	Industries []FilterOption `json:"industries"`
	States     []FilterOption `json:"states"`
	PriceRange PriceRange     `json:"price_range"`
}

type FilterOption struct {
	Value string `json:"value"`
	Label string `json:"label"`
	Count int    `json:"count"`
}

type PriceRange struct {
	Min int64 `json:"min"`
	Max int64 `json:"max"`
}

File: web/src/lib/types/listing.ts
export interface Listing {
	id: string;
	source_id: string;
	external_id: string;
	url: string;
	title: string;
	description?: string;
	asking_price?: number;
	revenue?: number;
	cash_flow?: number;
	ebitda?: number;
	inventory_value?: number;
	real_estate_included: boolean;
	real_estate_value?: number;
	city?: string;
	state?: string;
	zip_code?: string;
	country: string;
	lat?: number;
	lng?: number;
	industry?: string;
	industry_category?: string;
	business_type?: string;
	year_established?: number;
	employees?: number;
	reason_for_sale?: string;
	is_franchise: boolean;
	franchise_name?: string;
	first_seen_at: string;
	last_seen_at: string;
	is_active: boolean;
}

export interface ListingSearchParams {
	q?: string;
	price_min?: number;
	price_max?: number;
	revenue_min?: number;
	cash_flow_min?: number;
	states?: string[];
	industries?: string[];
	franchise?: boolean;
	real_estate?: boolean;
	bounds?: GeoBounds;
	sort?: string;
	page?: number;
	per_page?: number;
}

export interface GeoBounds {
	south_lat: number;
	west_lng: number;
	north_lat: number;
	east_lng: number;
}

export interface ListingSearchResult {
	listings: Listing[];
	total: number;
	page: number;
	per_page: number;
	total_pages: number;
}

export interface FilterOptions {
	industries: FilterOption[];
	states: FilterOption[];
	price_range: PriceRange;
}

export interface FilterOption {
	value: string;
	label: string;
	count: number;
}

export interface PriceRange {
	min: number;
	max: number;
}

export interface MapMarker {
	id: string;
	lat: number;
	lng: number;
	title: string;
	asking_price?: number;
	industry?: string;
}
